common/leica_sep.c

```diff
@@ -37,64 +45,104 @@ static int leica_sep_transfer(struct leica_sep_funcs *f, char *cmd, char *data,
 
 	f->puts(cmd);
 	cmd[1] = 'R';
-	*d++ = '-';
 	while(get_timer(0) < (start + timeout_ms))
 	{
 		if (!f->tstc())
 			continue;
 
-		data[bytes_received++] = f->getc();
-		*d++ = data[bytes_received -1];
+		data[bytes_received] = f->getc();
 
-		if ((data[bytes_received - 1] == '\n') || (bytes_received >= size)) {
-			if (bytes_received >= cmd_length &&
-			    !strncmp(data, cmd, cmd_length))
-			{
+		if ((data[bytes_received] == '\n') ||
+		    (bytes_received + 1 >= size)) {
+			if (bytes_received + 1 >= cmd_length &&
+			    !strncmp(data, cmd, cmd_length)) {
+				command_received = 1;
+				/* If \n is preceded by \r, we consider \r
+				 * as last character
+				 */
+				if (data[bytes_received - 1] != '\r')
+					bytes_received++;
+				break;
+			}
+			/* Enter here if we finish to read,
+			 * but we received a wrong response
+			 */
+			bytes_received = 0;
+			break;
+		} else if (data[bytes_received] == ':') {
+			column = 1;
+			bytes_received = 0;
+			continue;
+		} else if (column && data[bytes_received] == '"') {
+			quotes++;
+			if (quotes >= 2) {
```

Comment:
Why is this needed? The end of the message is marked with `\r\n`. Detecting quotes is not needed, right?
Start of payload: `if (data[bytes_received] == '"' && data[bytes_received - 1] == ':')`
End: `if (data[bytes_received] == '\n' && data[bytes_received - 1] == '\r' && data[bytes_received - 2] == '"')`
Or with `strncmp()`...